core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -0,0 +1,107 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.CreateNetworkCmd;
+import lombok.*;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TestRule;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.utility.ResourceReaper;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public interface Network extends AutoCloseable, TestRule {
+
+    String getId();
+
+    String getName();
+
+    Boolean getEnableIpv6();
+
+    String getDriver();
+
+    boolean isCreated();
+
+    default boolean create() {
+        return getId() != null;
```

Comment:
I would leave out the default implementation- it's too magic: you need to guess that getId is doing the actual create. 
Regarding the Boolean - why not fail if can't create? In most chances you can't continue without a network.